### ISSUE TITLE
TextLayer is not updated when the text is changed

### ISSUE THREAD

**Issue by txtl** [visgl/deck.gl#4987]
Hi, 

I have the codes below to show text changes. 

```
        var {
            TextLayer,
            Deck,
            OrthographicView,
        } = deck;

        const INITIAL_VIEW_STATE = {
            target: [0, 0, 0],
            zoom: 2,
        };

        const deckgl = new Deck({
            initialViewState: INITIAL_VIEW_STATE,
            controller: true,
            views: [new OrthographicView()],
            layers: [],
        });

        var testdata = [{
            position: [0, 0],
            number: "23",
            txtcolor: [255, 0, 0],
            txtsize: 12

        }]

        var getTextUpdated = 0;

        function _deckgl_render() {
            var textLayer = new TextLayer({
                id: "text-layer",
                data: testdata,
                getPosition: (d) => d.position,
                sizeUnits: "meters",
                fontWeight: "bold",
                fontFamily: "sans-serif",
                fontSettings: {
                    fontSize: 256,
                    buffer: 10,
                    radius: 8,
                },
                getText: (d) => {
                    console.log("getText: " + d.number);
                    return d.number
                },
                getSize: (d) => d.txtsize,
                getColor: (d) => d.txtcolor,
                updateTriggers: {
                    getText: getTextUpdated
                }
            });

            deckgl.setProps({
                layers: [textLayer]
            });
        }

        var toggle = 5;
        var last = 0;
        var now = 0;

        function animate() {
            now = Date.now();
            requestAnimationFrame(animate);
            if (now - last > 1000) {
                data = testdata[0];
                data.number = "" + (parseInt(data.number) + toggle);
                toggle *= -1;
                console.log("\n\ngetTextUpdated: " + getTextUpdated);
                console.log("updated to " + data.number + "");
                getTextUpdated = now;
                last = now;
               _deckgl_render();
            }
        }

        animate();
```
The text was not updated even if I have the updateTriggers specified for getText function. I may have missed something. I've scratched my head for hours :)

codepen to demonstrate https://codepen.io/txtl-smith/pen/oNxOgqG

I don't have this problem with IconLayer or PathLayer.  

Thank you,

**Comment by txtl** [comment]
If I change to 
```
updateTriggers: {
       all: getTextUpdated
}
```
The text is updated.

**Comment by Pessimistress** [comment]
I think this is a bug.

**Comment by Pessimistress** [comment]
Fix landed in 8.2.9

**Comment by txtl** [comment]
This is fixed. 

Thank you!
